### PR TITLE
Artifact download refactoring follow ups

### PR DESCRIPTION
**Pull request by vpodzime**
A follow-up for #93.

### CODE REVIEW

**Review by larsewi — APPROVED** [review]
Great, thanks 🚀

File: core/src/mender-artifact-download.c [hunk 1]
```diff
@@ -48,22 +48,18 @@ mender_download_artifact(const char *uri, mender_deployment_data_t *deployment_d
...
+        return ret;
     }
 
     /* Treatment depending of the status */
     if (200 == status) {
         /* Nothing to do */
-        ret            = MENDER_OK;
         *update_module = dl_data.update_module;
+        return MENDER_OK;
     } else {
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply core/src/mender-artifact-download.c:59]: It does, IMHO.

File: core/src/mender-artifact.c [hunk 2]
```diff
@@ -122,8 +122,14 @@ static mender_err_t artifact_read_data(mender_artifact_ctx_t *ctx, mender_artifa
...
-static mender_err_t process_artifact_data_callback(
-    const char *type, const cJSON *meta_data, const char *filename, size_t size, void *data, size_t index, size_t length, mender_artifact_download_data_t *dl_data);
+static mender_err_t process_artifact_data_callback(const char                      *type,
+                                                   const cJSON                     *meta_data,
+                                                   const char                      *filename,
+                                                   size_t                           size,
+                                                   void                            *data,
+                                                   size_t                           index,
+                                                   size_t                           length,
+                                                   mender_artifact_download_data_t *dl_data);
```
Comment (lluiscampos) [inline core/src/mender-artifact.c:132]: The formatting (here and below) belongs to an earlier commit ;)

File: core/src/mender-artifact.c [hunk 3]
```diff
@@ -1035,6 +1035,18 @@ artifact_read_meta_data(mender_artifact_ctx_t *ctx) {
...
+/**
+ * @brief Callback function to be invoked to perform the treatment of the data from the artifact
+ * @param type Type from header-info payloads
+ * @param meta_data Meta-data from header tarball
+ * @param filename Artifact filename
+ * @param size Artifact file size
+ * @param data Artifact data
+ * @param index Artifact data index
+ * @param length Artifact data length
+ * @param dl_data Download data for the artifact
```
Comment (lluiscampos) [inline core/src/mender-artifact.c:1047]: This is not the "Artifact" but the payload, right?

Comment (lluiscampos) [reply core/src/mender-artifact.c:1047]: Not even the payload, but one file inside the payload :thinking:

**Review by lluiscampos — APPROVED** [review]

File: core/src/mender-artifact.c (hunk 3 above)
Comment (vpodzime) [reply core/src/mender-artifact.c:1047]: It's "artifact download data", so... :man_shrugging:

File: core/src/mender-artifact.c (hunk 2 above)
Comment (vpodzime) [reply core/src/mender-artifact.c:132]: Good catch!

File: core/src/mender-artifact.c (hunk 3 above)
Comment (vpodzime) [reply core/src/mender-artifact.c:1047]: Or meta-data, actually. But I called the variable `dl_data` everywhere and the type is also called like that, so I'll stick to it. At least for now.

**Pull request merged**